config/src/main/java/org/springframework/security/config/annotation/method/configuration/AuthorizationProxyWebConfiguration.java

```diff
@@ -16,27 +16,41 @@
 
 package org.springframework.security.config.annotation.method.configuration;
 
+import java.util.List;
 import java.util.Map;
 
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
 import org.springframework.beans.factory.config.BeanDefinition;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Role;
 import org.springframework.http.HttpEntity;
 import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotWritableException;
+import org.springframework.security.access.AccessDeniedException;
 import org.springframework.security.authorization.method.AuthorizationAdvisorProxyFactory;
+import org.springframework.security.web.util.ThrowableAnalyzer;
+import org.springframework.web.servlet.HandlerExceptionResolver;
 import org.springframework.web.servlet.ModelAndView;
 import org.springframework.web.servlet.View;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
 
 @Configuration
-class AuthorizationProxyWebConfiguration {
+class AuthorizationProxyWebConfiguration implements WebMvcConfigurer {
 
 	@Bean
 	@Role(BeanDefinition.ROLE_INFRASTRUCTURE)
 	AuthorizationAdvisorProxyFactory.TargetVisitor webTargetVisitor() {
 		return new WebTargetVisitor();
 	}
 
+	@Override
+	public void extendHandlerExceptionResolvers(List<HandlerExceptionResolver> resolvers) {
+		resolvers.add(0, new HttpMessageNotWritableAccessDeniedExceptionResolver());
```

Comment:
One thing that concerns me is that this might break applications that have an `@ExceptionHandler` for `AccessDeniedException`-caused `HttpMessageNotWritableException` already.

Further, it would be nice if there were a clear way for applications to override Security's default handling of this scenario.

Reply:
I implemented a fix for this scenario by finding the `DefaultHandlerExceptionResolver`'s index and adding Security's `ExceptionResolver` before the default resolver.  This way user-defined `@ExceptionHandler` will always take precedence and try to resolve an exception first since the `ExceptionHandlerExceptionResolver` is added before the `extendHandlerExceptionResolvers` method call. I think we could also consider providing a configuration option to control wether Spring Security should handle this type of scenario or not.

Reply:
I imagine that the way to override would be to add an exception handler or add their own exception resolver. Is there another way that you had in mind?

Reply:
I think then it's possible now to override it by defining a custom '@ExceptionHandler', I have added a test covering that.